hkmc2/shared/src/main/scala/hkmc2/codegen/Lifter.scala

```diff
@@ -613,6 +613,12 @@ class Lifter(handlerPaths: Opt[HandlerPaths])(using State, Raise):
         case Assign(t: TermSymbol, rhs, rest) if t.owner.isDefined =>
           ctx.getIsymPath(t.owner.get) match
             case Some(value) if !belongsToCtor(t.owner.get) =>
+              if (t.k is syntax.LetBind) && !t.owner.forall(_.isInstanceOf[semantics.TopLevelSymbol]) then
+                // TODO: can we do better?
```

Comment:
It's possible that we can ignore nested functions/classes which reference private variables, but this will make the lifter even more complicated than it already is. For now I think this is acceptable.

Reply:
This is just for showing something when a private variable is encountered instead of just failing silently. Not meant to be fixing private variables in this PR.

Reply:
```suggestion
                // TODO: improve the error message
```

Reply:
@AnsonYeung it looks like you missed my suggestion when marking this conversation as "resolved".